lib/config/wrapper.go

```diff
@@ -281,17 +275,15 @@ func (w *Wrapper) SetFolder(fld FolderConfiguration) (Waiter, error) {
 	defer w.mut.Unlock()
 
 	newCfg := w.cfg.Copy()
-	replaced := false
+
 	for i := range newCfg.Folders {
 		if newCfg.Folders[i].ID == fld.ID {
-			newCfg.Folders[i] = fld
-			replaced = true
-			break
+			newCfg.Folders[i] = fld.Copy()
+			return w.replaceLocked(newCfg)
 		}
 	}
-	if !replaced {
-		newCfg.Folders = append(w.cfg.Folders, fld)
```

Comment:
So after a bunch of bisecting, this seems to be an issue, essentially appending new folders to a copy from the old map, yet I don't understand why this could be an issue, as all of this happens under a lock.

Reply:
I'm just shooting from the hip here, but could there be an issue when two folders are added in rapid successions, such as from the same cluster config. That is, if the folder list is initially A, B and we append to A, B, C but this doesn't get propagated and the next folder added gets appended to A, B as A, B, D - The "C" folder is now mysteriously gone and bad things happen...

Reply:
I don't have a concrete scenario, Both auto accepting happening in `ClusterConfig` and `CommitConfiguration` happen without lock on `fmut`. And `CommitConfiguration` is called in a goroutine, i.e. the wrapper lock won't be held while the model is running it. So in principle it might be possible for `CommitConfiguration` removing/adding folders at the same time as `ClusterConfig` does.